activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -223,17 +223,17 @@ def operation_over_aggregate_column(column, operation, distinct)
       end
 
       def execute_simple_calculation(operation, column_name, distinct) #:nodoc:
-        # PostgreSQL doesn't like ORDER BY when there are no GROUP BY
-        relation = unscope(:order)
```

Comment:
@kamipo @rafaelfranca This change has caused an unexpected SQL error when a query contains both `.distinct` and `.limit`/`.offset`:

```
ActiveRecord::StatementInvalid: PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list
```

In most SQL servers (e.g. PostSQL, SQL Server etc), columns listed in DISTINCT must also be listed in ORDER BY. This simple query that used to run in Rails 5.0 will now fail:

```
Model.distinct.order(updated_at: :desc).offset(10).count
```
because "count" operation adds column to DISTINCT list but not unscoping existing ORDER BY:

```
SELECT
    COUNT(DISTINCT count_column)
FROM (
    SELECT
        DISTINCT "models"."id" AS count_column
    FROM
        "models"
    ORDER BY
        "models"."updated_at" DESC  /* Error, column list not same as DISTINCT */
    LIMIT $1) subquery_for_count
```

See issues reported by others: https://github.com/kaminari/kaminari/issues/888, https://github.com/spark-solutions/spree/commit/b5785add448062d7a7ed125ae4c5556f060a556e